db/revtree.go

```diff
@@ -783,7 +784,7 @@ func encodeRevisions(revs []string) Revisions {
 		if i == 0 {
 			start = gen
 		} else if gen != start-i {
-			base.Warnf("Found gap in revision list. Expecting gen %v but got %v in %v", start-i, gen, revs)
+			base.WarnfCtx(ctx, "Found gap in revision list. Expecting gen %v but got %v in %v", start-i, gen, revs)
```

Comment:
I haven't done a complete trace, but it seems like there are several places we're forced to pass context because we end up here.  This particular warning I think would benefit more from docID than user context, since it's primarily informational and on its own doesn't indicate a particular issue.

Reply:
Agreed. Will swap context for docID and cleanup obsolete context in callers.